src/System.Device.Gpio/System/Device/IQueryComponentInformation.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device
+{
+    /// <summary>
+    /// Interface to support querying component information from a class
+    /// </summary>
+    public interface IQueryComponentInformation
+    {
+        /// <summary>
+        /// Query information about a component and it's children.
+        /// </summary>
+        /// <param name="onlyActive">True to return only active components, false to also list possible alternatives or inactive drivers</param>
```

Comment:
not sure I love `onlyActive`, should it instead be something like `listAllComponents`?

Reply:
As discussed: It will be better to have a separate method for that.